lib/ApplicationFeatures/ApplicationServer.h

```diff
@@ -229,6 +231,11 @@ class ApplicationServer {
   // signal the server to shut down
   void beginShutdown();
 
+  // set state to unpreparing phase - no more waiting for external resources from here.
+  void beginUnprepare() {
+    _unpreparing.exchange(true);
```

Comment:
This is counter-intuitive, as `_unpreparing` here does not mean the server is in the UNPREPARED state.